File: Example/PrebidDemoJava/src/main/java/org/prebid/mobile/javademo/testcases/TestCaseAdapter.java
package org.prebid.mobile.javademo.testcases;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.prebid.mobile.javademo.databinding.ListItemAdTypeBinding;

import java.util.ArrayList;
import java.util.List;

public class TestCaseAdapter extends RecyclerView.Adapter<TestCaseAdapter.AdTypeViewHolder> {

    private List<TestCase> list = new ArrayList<>();
    private final OnItemClick onItemClick;

    public TestCaseAdapter(OnItemClick onItemClick) {
        this.onItemClick = onItemClick;
    }

    @NonNull
    @Override
    public AdTypeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ListItemAdTypeBinding binding = ListItemAdTypeBinding.inflate(
            LayoutInflater.from(parent.getContext()),
            parent,
            false
        );

        AdTypeViewHolder viewHolder = new AdTypeViewHolder(binding);
        viewHolder.setOnItemClickedListener(onItemClick);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull AdTypeViewHolder holder, int position) {
        holder.bind(list.get(position));
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setList(List<TestCase> list) {
        this.list = list;
        notifyDataSetChanged();
    }

    static class AdTypeViewHolder extends RecyclerView.ViewHolder {

        private final ListItemAdTypeBinding binding;
        private TestCase testCase;

        AdTypeViewHolder(ListItemAdTypeBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        void bind(TestCase testCase) {
            this.testCase = testCase;
            binding.tvName.setText(itemView.getContext().getString(testCase.getTitleStringRes()));
        }

        void setOnItemClickedListener(OnItemClick onItemClickListener) {
            binding.getRoot().setOnClickListener(v -> {
                onItemClickListener.click(testCase);
            });
        }

    }

    public interface OnItemClick {

        void click(TestCase testCase);

    }

}

File: Example/PrebidDemoKotlin/src/main/java/org/prebid/mobile/prebidkotlindemo/testcases/TestCaseAdapter.kt
package org.prebid.mobile.prebidkotlindemo.testcases

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.prebid.mobile.prebidkotlindemo.databinding.ListItemAdTypeBinding

class TestCaseAdapter(
    private val onItemClicked: (TestCase) -> Unit
) : RecyclerView.Adapter<TestCaseAdapter.AdTypeViewHolder>() {

    private var list: List<TestCase> = arrayListOf()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AdTypeViewHolder {
        val binding = ListItemAdTypeBinding.inflate(
            LayoutInflater.from(parent.context),
            parent,
            false
        )

        val viewHolder = AdTypeViewHolder(binding)
        viewHolder.setOnItemClickedListener(onItemClicked)
        return viewHolder
    }

    override fun onBindViewHolder(holder: AdTypeViewHolder, position: Int) {
        holder.bind(list[position])
    }

    override fun getItemCount() = list.size


    @SuppressLint("NotifyDataSetChanged")
    fun setList(list: List<TestCase>) {
        this.list = list
        notifyDataSetChanged()
    }


    class AdTypeViewHolder(
        private val binding: ListItemAdTypeBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        private var testCase: TestCase? = null

        fun bind(testCase: TestCase) {
            this.testCase = testCase
            binding.tvName.text = itemView.context.getString(testCase.titleStringRes)
        }

        fun setOnItemClickedListener(onItemClicked: (TestCase) -> Unit) {
            binding.root.setOnClickListener {
                onItemClicked(testCase ?: return@setOnClickListener)
            }
        }

    }

}